scripts/qiita-plugin-launcher

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from subprocess import Popen
+
+import click
+
+
+@click.command()
+@click.argument('env_script', required=True)
+@click.argument('start_script', required=True)
+@click.argument('url', required=True)
+@click.argument('job_id', required=True)
+@click.argument('job_dir', required=True)
+def start(env_script, start_script, url, job_id, job_dir):
+    """Starts the plugin environment"""
+    cmd = [start_script, url, job_id, job_dir]
+    # When Popen executes, the shell is not in interactive mode,
+    # so it is not sourcing any of the bash configuration files
+    # We need to source it so the env_script are available
+    cmd = "source ~/.bash_profile; %s; %s" % (env_script, ' '.join(cmd))
```

Comment:
Got it, would be useful to note this somewhere in the documentation of
this script.

On (Dec-04-15|13:59), Jose Navas wrote:
> > +import click
> > +
> > +
> > +@click.command()
> > +@click.argument('env_script', required=True)
> > +@click.argument('start_script', required=True)
> > +@click.argument('url', required=True)
> > +@click.argument('job_id', required=True)
> > +@click.argument('job_dir', required=True)
> > +def start(env_script, start_script, url, job_id, job_dir):
> > +    """Starts the plugin environment"""
> > +    cmd = [start_script, url, job_id, job_dir]
> > +    # When Popen executes, the shell is not in interactive mode,
> > +    # so it is not sourcing any of the bash configuration files
> > +    # We need to source it so the env_script are available
> > +    cmd = "source ~/.bash_profile; %s; %s" % (env_script, ' '.join(cmd))
> 
> The script `qiita-plugin-launcher` is just an example on how they have to look and it is related to the comment that you made above about adding admin documentation. For example, this script will be different in barnacle.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1558/files#r46737562